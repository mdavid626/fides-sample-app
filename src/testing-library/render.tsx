import React from 'react';
import { createBrowserRouter, RouterProvider } from 'react-router-dom';
import { render } from '@testing-library/react';

export const renderWithRouter = (component: JSX.Element) => {
  const router = {} as { current?: ReturnType<typeof createBrowserRouter> };
  return {
    ...render(component, {
      wrapper: ({ children }) => {
        const browserRouter = createBrowserRouter([
          {
            path: '',
            element: children,
          },
        ]);
        router.current = browserRouter;
        return <RouterProvider router={browserRouter} />;
      },
    }),
    router,
  };
};
