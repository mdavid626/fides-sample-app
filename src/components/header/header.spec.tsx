import React from 'react';
import { renderWithRouter } from '../../testing-library/render';
import Header from './header';

describe('header', () => {
  it('should render', () => {
    const { asFragment } = renderWithRouter(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });
});
