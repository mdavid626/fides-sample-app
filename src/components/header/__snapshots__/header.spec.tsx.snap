// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header should render 1`] = `
<DocumentFragment>
  <div
    class="Header"
  >
    <div
      class="Header-title"
    >
      Fides Sample App
    </div>
    <div
      class="Header-navigation"
    >
      <a
        class="Header-link Header-selectedLink"
        href="/"
      >
        Movies
      </a>
      <a
        class="Header-link"
        href="/favourites"
      >
        Favourites
      </a>
      <a
        class="Header-link"
        href="/about"
      >
        About
      </a>
    </div>
  </div>
</DocumentFragment>
`;
